src/map/clif.c

```diff
@@ -10585,6 +10585,45 @@ static void clif_parse_LoadEndAck(int fd, struct map_session_data *sd)
 		clif->updatestatus(sd, SP_CARTINFO);
 	}
 
+	/**
+	 * In official servers, an item's unequip script is executed when entering a zone where the item is restricted,
+	 * even if the item won't be unequipped.
+	 *
+	 **/
+	for (int i = 0; i < EQI_MAX; i++) {
+		if (sd->equip_index[i] >= 0) {
+			struct item_data *equip_data = sd->inventory_data[sd->equip_index[i]];
```

Comment:
Actually this should be impossible, since the item is equipped, but I added a `NULL` validation.
Additionally I added a validation for item which take multiple equipment slots, do prevent running the same unequip script multiple times.